File: src/main/java/com/mallang/auth/query/MemberQueryService.java
package com.mallang.auth.query;

import com.mallang.auth.domain.Member;
import com.mallang.auth.query.repository.MemberQueryRepository;
import com.mallang.auth.query.response.MemberResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@RequiredArgsConstructor
@Transactional(readOnly = true)
@Service
public class MemberQueryService {

    private final MemberQueryRepository memberQueryRepository;

    public MemberResponse findProfile(Long memberId) {
        Member member = memberQueryRepository.getById(memberId);
        return MemberResponse.from(member);
    }
}

File: src/main/kotlin/com/mallang/auth/query/MemberQueryService.kt
package com.mallang.auth.query

import com.mallang.auth.query.dao.MemberProfileDataDao
import com.mallang.auth.query.dao.model.MemberProfileQueryModel
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional


@Transactional(readOnly = true)
@Service
class MemberQueryService(
        private val memberProfileDataDao: MemberProfileDataDao
) {

    fun findProfile(memberId: Long): MemberProfileQueryModel {
        return memberProfileDataDao.find(memberId)
    }
}
